### PR TITLE
feat: add formatted logging to file for internal

### LINKED ISSUE #501: Logging for Internal
### Is your feature request related to a problem? Please describe.

We should be logging into a file and change our `fmt.Print`'s to `log`

### Describe the solution you'd like

Just some basic logging, so we can trace errors

### Describe alternatives you've considered

_No response_

### Additional context

Inspiration:
https://www.honeybadger.io/blog/golang-logging/

### PR DESCRIPTION
**Pull request by nullxDEADBEEF**
# Description

Please include a summary of the change and which issue is fixed.

Adds logging info to a file, to easily trace errors.

Fixes #501

### CODE REVIEW

File: internal/routes.go [hunk 1]
```diff
@@ -249,7 +249,7 @@ func fetchPerson(id string, personCh chan Person) {
 	person := Person{}
 	errDecode := json.NewDecoder(res.Body).Decode(&person)
 	if errDecode != nil {
-		fmt.Println("Failed to decode person: ", errDecode, " with ID: ", id)
+		log.Println("Failed to decode person: ", errDecode, " with ID: ", id)
```
Comment (AndreasPB) [inline internal/routes.go:252]: I think we should add how we react to the error as well, as in the weird `Id : -1` skip hack. I think I wrote ` - Will skip`

File: .gitignore [hunk 2]
```diff
@@ -39,5 +39,6 @@ frontend/.svelte-kit
 /letsencrypt
 backend/json.gz_dumps
 internal/tmp
+internal/logs.txt
```
Comment (AndreasPB) [inline .gitignore:42]: Does this file get updated outside of the container when working with it locally?

Comment (nullxDEADBEEF) [reply .gitignore:42]: yep, it gets updated outside the container

File: internal/routes.go (hunk 1 above)
Comment (nullxDEADBEEF) [reply internal/routes.go]: Done

File: internal/routes.go [hunk 3]
```diff
@@ -74,7 +74,7 @@ func fetchMedia(c *gin.Context) (chan Movie, chan TV, chan Person) {
 				<-guardCh
 			}(id, personCh)
 		default:
-			log.Fatal("This is not the movie type you seek")
+			log.Fatal("This is not the movie type you seek - skipping to next media")
```
Comment (AndreasPB) [inline internal/routes.go:77]: This wont skip, it's just break and die :D Fatal consists of an `os.Exit(1)` https://cs.opensource.google/go/go/+/refs/tags/go1.20.4:src/log/log.go;l=363

It needs the weird `Id: -1`-hack if we want it to skip

Comment (AndreasPB) [reply internal/routes.go:77]:
```suggestion
			log.Println("This is not the movie type you seek - skipping to next media")
```

File: .gitignore (hunk 2 above)
Comment (AndreasPB) [reply .gitignore:42]: Sweet!

File: internal/main.go [hunk 4]
```diff
@@ -18,6 +19,16 @@ func main() {
...
 
+	log.SetFlags(log.Ldate | log.Ltime | log.Lshortfile)
+
+	// permission mode 0666 -> all users can read and write but cannot execute file
+	logFile, err := os.OpenFile("logs.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		log.Println("there was an error with logs.txt: ", err)
+	}
+
+	log.SetOutput(logFile)
```
Comment (AndreasPB) [inline internal/main.go:30]: Is it possible to have multiple outputs? It seems like it isn't logging to the terminal anymore

Comment (AndreasPB) [reply internal/main.go:30]: Also, should the file be closed (maybe with defer)?

Comment (nullxDEADBEEF) [reply internal/main.go]: Done

Comment (AndreasPB) [reply internal/main.go]: Sick

**Review by AndreasPB — APPROVED** [review]

**Pull request merged**